test/semantics/doconcurrent06.f90

```diff
@@ -0,0 +1,44 @@
+! Copyright (c) 2019, Arm Ltd.  All rights reserved.
+!
+! Licensed under the Apache License, Version 2.0 (the "License");
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+
+! RUN: ${F18} -funparse-with-symbols %s 2>&1 | ${FileCheck} %s
+! CHECK: exit from DO CONCURRENT construct
+! CHECK: exit from DO CONCURRENT construct
+! CHECK: exit from DO CONCURRENT construct
+! CHECK: exit from DO CONCURRENT construct
+
+subroutine do_concurrent_test1(n)
+  implicit none
+  integer :: j,k,n
+  mytest: if (n>0) then
+  mydoc:    do concurrent(j=1:n)
+              do k=1,n
+                if (j==10) exit mytest
+              end do
+            end do mydoc
+          end if mytest
+end subroutine do_concurrent_test1
+
+subroutine do_concurrent_test2(n)
+  implicit none
+  integer :: j,k,n
+  mytest: if (n>0) then
+  mydoc:    do concurrent(j=1:n)
+              do concurrent(k=1:n)
+                if (k==5) exit
+                if (j==10) exit mytest
+              end do
+            end do mydoc
+          end if mytest
+end subroutine do_concurrent_test2
```

Comment:
Should we have a test that interleaves a nest of DO and DO CONCURRENT loops, say, up to 4 or 6 total?

Reply:
I have added some tests that interleave do and do concurrent in test/semantics/doconcurrent07.f90